Mathlib/Combinatorics/SimpleGraph/Clique.lean

```diff
@@ -213,13 +213,27 @@ section DecidableEq
 
 variable [DecidableEq α]
 
-theorem IsNClique.insert (hs : G.IsNClique n s) (h : ∀ b ∈ s, G.Adj a b) :
+protected theorem IsNClique.insert (hs : G.IsNClique n s) (h : ∀ b ∈ s, G.Adj a b) :
     G.IsNClique (n + 1) (insert a s) := by
   constructor
   · push_cast
     exact hs.1.insert fun b hb _ => h _ hb
   · rw [card_insert_of_not_mem fun ha => (h _ ha).ne rfl, hs.2]
 
+lemma IsNClique.erase_of_mem (hs : G.IsNClique n s) (ha : a ∈ s) :
+    G.IsNClique (n - 1) (s.erase a) := by
+  constructor
+  · apply hs.1.subset; simp
+  · rw [card_erase_of_mem ha, hs.2]
+
+lemma IsNClique.insert_erase (hs : G.IsNClique n s) (ha : ∀ w ∈ s, w ≠ b → G.Adj a w) (hb : b ∈ s) :
```

Comment:
Two minor points:

- Let's make this `protected` since there also exists `Finset.insert_erase`
- I think the below spelling of `ha` is slightly preferable
```suggestion
protected lemma IsNClique.insert_erase
    (hs : G.IsNClique n s) (ha : ∀ w ∈ s \ {b}, G.Adj a w) (hb : b ∈ s) :
```